lib/hanami/events/subscriber.rb

```diff
@@ -1,19 +1,24 @@
 module Hanami
   module Events
     class Subscriber
-      def initialize(pattern, event_handler)
+      def initialize(pattern, event_handler, logger = nil)
+        @logger = logger
         @pattern = pattern
         @pattern_matcher = Matcher.new(pattern)
         @event_handler = event_handler
       end
 
       def call(event_name, payload)
-        @event_handler.call(payload) if @pattern_matcher.match?(event_name)
+        instance_exec(payload, &@event_handler) if @pattern_matcher.match?(event_name)
```

Comment:
By using `instance_exec` we expose all internal state of subscriber to event handler, i.e. they can get an access to `pattern`, `pattern_matcher` in `event_handler` block. I'm thinking about the other pay of passing logger to handler...